### PR TITLE
#126: Send notification and an email when a lesson has been cancelled

### PR DESCRIPTION
**Pull request by AhmedNSidd**
Addresses Issue #126 by sending a mail and a notification when a lesson is cancelled. If a lesson is cancelled when it was in the pending state by the person who created it, then no notification is sent out.

### CODE REVIEW

File: schedule_lessons/dashboard/views.py [hunk 1]
```diff
@@ -973,20 +992,47 @@ def error_check_and_save_lesson(request, lesson, context):
...
-        context["status"] = 200
         end_time_in_local_time = datetime.datetime.combine(date, end_time,
                                                            time_difference)
         lesson.start_time = start_time_in_local_time.astimezone(UTC_ZONE)
         lesson.end_time = end_time_in_local_time.astimezone(UTC_ZONE)
+        availabilities = list(Availability.objects.filter(
+                                    profile__user=person_to_schedule_with,
+                                    day=lesson.start_time.strftime("%A")))
+        context['non_available_time_error'] = True
+        if availabilities:
```
Comment (sdevalapurkar) [inline schedule_lessons/dashboard/views.py]: do we need this check? if availabilities is empty or null wont the for loop below just not execute?

File: schedule_lessons/dashboard/views.py [hunk 2]
```diff
@@ -973,20 +992,47 @@ def error_check_and_save_lesson(request, lesson, context):
...
+            for availability in availabilities:
+                lesson_start_t = lesson.start_time.time()
+                availability_start_t = availability.start_time.time()
+                lesson_end_t = lesson.end_time.time()
+                availability_end_t = availability.end_time.time()
+                if ((lesson_end_t < lesson_start_t and
```
Comment (sdevalapurkar) [inline schedule_lessons/dashboard/views.py]: holy crap this conditional

File: schedule_lessons/dashboard/views.py (hunk 1 above)
Comment (AhmedNSidd) [reply schedule_lessons/dashboard/views.py]: Yeah you're right

File: schedule_lessons/dashboard/views.py (hunk 2 above)
Comment (AhmedNSidd) [reply schedule_lessons/dashboard/views.py]: Yup, there's no getting around it. Working with timings is a bitch

Comment (sdevalapurkar) [reply schedule_lessons/dashboard/views.py]: that sucks, but yeah looks good man

**Pull request merged**